include/triton/Dialect/TritonNvidiaGPU/IR/TritonNvidiaGPUOps.td

```diff
@@ -316,7 +316,7 @@ def TTNG_AsyncTMACopyLocalToGlobalOp : TTNG_Op<"async_tma_copy_local_to_global">
   }];
 
   let arguments = (ins
-    Arg<TT_PtrType, "", [MemWrite<GlobalMemory>]>:$desc_ptr,
+    Arg<TT_PtrType, "", [MemRead<GlobalMemory>, MemWrite<GlobalMemory>]>:$desc_ptr,
```

Comment:
why is this a read here?

Reply:
The TMA engine reads information out of the descriptor. At the same time, it writes to an address stored inside of it.

Reply:
ah good point. Note for later I wonder if we will want to separate out the descriptor memory in its own memory space in TTGIR as it is not something exposed to user like global memory is and live in its own scratch memory/constant memory.

Reply:
Yes I talked to Peter about that. I think it would be better to model it separately. On that topic, we probably also want separate memory spaces for things like the TC instruction pipeline, etc. for really fine grained side effect modeling.